tools/buildgen/generate_projects.sh

```diff
@@ -18,9 +18,19 @@ set -e
 
 export TEST=${TEST:-false}
 
+gen_build_files=""
+function cleanup {
+  rm $gen_build_files
+}
+trap cleanup EXIT
+
 echo "Generating build_autogenerated.yaml from bazel BUILD file"
 rm -f build_autogenerated.yaml
 python3 tools/buildgen/extract_metadata_from_bazel_xml.py
+if [ $? -ne 0 ]; then
```

Comment:
Does `set -e` do the job? Do you want to make this more explicit? I'm okay with both ways but it may look redundant.

Reply:
Ah, you're right, these status code checks are unreachable with `set -e`. Removed.